L1Trigger/L1THGCal/src/HGCalTriggerTowerGeometryHelper.cc

```diff
@@ -0,0 +1,134 @@
+
+#include "L1Trigger/L1THGCal/interface/HGCalTriggerTowerGeometryHelper.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include "FWCore/Utilities/interface/EDMException.h"
+#include "DataFormats/ForwardDetId/interface/HGCalDetId.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include <cmath>
+#include <iostream>
+#include <fstream>
+#include <algorithm>
+
+
+HGCalTriggerTowerGeometryHelper::HGCalTriggerTowerGeometryHelper(const edm::ParameterSet& conf) : minEta_(conf.getParameter<double>("minEta")),
+                                                                                                  maxEta_(conf.getParameter<double>("maxEta")),
+                                                                                                  minPhi_(conf.getParameter<double>("minPhi")),
+                                                                                                  maxPhi_(conf.getParameter<double>("maxPhi")),
+                                                                                                  nBinsEta_(conf.getParameter<int>("nBinsEta")),
+                                                                                                  nBinsPhi_(conf.getParameter<int>("nBinsPhi")),
+                                                                                                  binsEta_(conf.getParameter<std::vector<double> >("binsEta")),
+                                                                                                  binsPhi_(conf.getParameter<std::vector<double> >("binsPhi")) {
+
+
+  if(!binsEta_.empty() && ((unsigned int)(binsEta_.size()) != nBinsEta_+1)) {
+    throw edm::Exception(edm::errors::Configuration, "Configuration")
+      << "HGCalTriggerTowerGeometryHelper nBinsEta for the tower map not consistent with binsEta size"<<std::endl;
```

Comment:
why not just replace the value of `nBinsEta_` with `binsEta_.size()` if `binsEta_` is specified?

Reply:
@kpedro88: I don't particularly like overriding user parameters. I think it is better to enforce consistency. Note however that this was already the behavior of the previous version of the code.